src/proxy/http/metrics/mod.rs

```diff
@@ -42,13 +45,31 @@ pub trait Stats {
 }
 
 #[derive(Debug)]
-pub struct RequestMetrics<C>
+pub struct RequestMetrics<R, C>
 where
+    R: Hash + Eq,
     C: Hash + Eq,
 {
     last_update: Instant,
-    total: Counter,
     by_retry_skipped: IndexMap<RetrySkipped, Counter>,
+    by_dst: IndexMap<Option<Addr>, DstMetrics<R, C>>,
```

Comment:
I don't really understand why the metrics registry has to be aware of any of this... It seems like a it should all be a detail of the stack targets...

Do we not have both the logical and concrete dsts in the endpoint stack target? If not, can we?

Reply:
We don't have the logical and concrete dsts in the endpoint stack target, just the concrete.  We could have the logical dst in there, but this would make it so that concrete dst load balancers would not be shared between logical dsts.  

This is why we propagate the logical dst in the request extensions.  Having the logical dst in the target of the endpoint stack would certainly simplify this a lot.  Not sure how to evaluate the tradeoff, though.

Reply:
Per offline discussion: because a concrete lb is not shared across multiple logical names, we actually can have all of the needed metadata in the endpoint stack's target, which should greatly simplify the change.